easybuild/framework/easyblock.py

```diff
@@ -1812,10 +1814,16 @@ def check_checksums_for(self, ent, sub='', source_cnt=None):
             msg += "vs %d checksums" % checksum_cnt
             checksum_issues.append(msg)
 
-        for fn, checksum in zip(sources + patches, checksums):
-            if not is_sha256_checksum(checksum):
-                msg = "Non-SHA256 checksum found for %s: %s" % (fn, checksum)
-                checksum_issues.append(msg)
+        if isinstance(checksums, (tuple, list)):
+            for fn, checksum in zip(sources + patches, checksums):
+                if not is_sha256_checksum(checksum):
+                    msg = "Non-SHA256 checksum found for %s: %s" % (fn, checksum)
+                    checksum_issues.append(msg)
+        elif isinstance(checksums, dict):
+            for fn, checksum in checksums.items():
+                if not is_sha256_checksum(checksum):
+                    msg = "Non-SHA256 checksum found for %s: %s" % (fn, checksum)
+                    checksum_issues.append(msg)
```

Comment:
This doesn't look correct...

`checksums` here is the full list of checksums, so it should be something like:

```python
for fn, checksum in zip(sources + patches, checksums):
    if isinstance(checksum, basestring):
        if not is_sha256_checksum(checksum):
            ...
    elif isinstance(checksum, dict):
        non_sha256_checksums = [(fn, c) for (fn, c) in checksum.values() if is_sha256_checksum(c)]
        if non_sha256_checksums:
            msg = "Non-SHA256 checksums found: %s" % non_sha256_checksums
            checksum_issues.append(msg)
```

(probably needs a bit more tweaking for proper formatting)